vault/activity_log.go

```diff
@@ -377,22 +376,10 @@ func (a *ActivityLog) saveCurrentSegmentInternal(ctx context.Context, force bool
 			return err
 		}
 
-		// Compress the contents of the segment. compressutil uses a prepended
-		// canary to the encoded message. This way the decompression can be made
-		// backwards compatible. This works by checking the presence of
-		// canaries; absence of canaries will fallback to processing
-		// uncompressed data.
-		compressedClients, err := compressutil.Compress(clients, &compressutil.CompressionConfig{
```

Comment:
We are also using compression to store the precomputed queries. Even that should be removed.

Reply:
But they're JSON, no?

Reply:
Ah, yes. They are.